src/PlusDataCollection/PointGrey/vtkPlusSpinnakerVideoSource.cxx

```diff
@@ -0,0 +1,807 @@
+    this->Internal->SystemPtr->ReleaseInstance();
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Failed to de-initialize camera. Exception text: " << e.what());
+    return PLUS_FAIL;
+  }
+
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::InternalStartRecording()
+{
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::InternalStartRecording()");
+
+  try
+  {
+    // get camera nodemap pointer
+    Spinnaker::GenApi::INodeMap & nodeMapTLDevice = this->Internal->CameraPtr->GetTLDeviceNodeMap();
+    Spinnaker::GenApi::INodeMap& nodeMap = this->Internal->CameraPtr->GetNodeMap();
+
+    // set acquisition mode to continuous
+    Spinnaker::GenApi::CEnumerationPtr ptrAcquisitionMode = nodeMap.GetNode("AcquisitionMode");
+    if (!IsAvailable(ptrAcquisitionMode) || !IsWritable(ptrAcquisitionMode))
+    {
+      LOG_ERROR("Unable to set acquisition mode to continuous (node retrieval). Aborting...");
+      return PLUS_FAIL;
+    }
+    Spinnaker::GenApi::CEnumEntryPtr ptrAcquisitionModeContinuous = ptrAcquisitionMode->GetEntryByName("Continuous");
+    if (!IsAvailable(ptrAcquisitionModeContinuous) || !IsReadable(ptrAcquisitionModeContinuous))
+    {
+      LOG_ERROR("Unable to set acquisition mode to continuous (entry 'continuous' retrieval). Aborting...");
+      return PLUS_FAIL;
+    }
+    int64_t acquisitionModeContinuous = ptrAcquisitionModeContinuous->GetValue();
+    ptrAcquisitionMode->SetIntValue(acquisitionModeContinuous);
+    LOG_INFO("Acquisition mode set to continuous...");
+
+    // Begin acquiring images
+    this->Internal->CameraPtr->BeginAcquisition();
+
+    std::string deviceSerialNumber("");
+    Spinnaker::GenApi::CStringPtr ptrStringSerial = nodeMapTLDevice.GetNode("DeviceSerialNumber");
+    if (IsAvailable(ptrStringSerial) && IsReadable(ptrStringSerial))
+    {
+      deviceSerialNumber = ptrStringSerial->GetValue();
+      LOG_INFO("Device serial number retrieved as " << deviceSerialNumber << ".");
+    }
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Failed to start recording. Exception text: " << e.what());
+    return PLUS_FAIL;
+  }
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::InternalStopRecording()
+{
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::InternalStopRecording()");
+
+  try
+  {
+    // End acquiring images
+    this->Internal->CameraPtr->EndAcquisition();
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Failed to end recording. Exception text: " << e.what());
+    return PLUS_FAIL;
+  }
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::NotifyConfigured()
+{
+  // TODO: Implement some configuration checks here
```

Comment:
Please implement. Some basic checks like number of video sources >= or == 1, number of output channels >= or == 1, etc... would be fine

Reply:
Ah okay - it was not clear to me what should go in this method. Perhaps we should write a documentation page about the standard device methods for a subclass of vtkPlusDevice?